compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5525,9 +5525,37 @@ TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, i
     TR::MemoryReference *tempMR = TR::MemoryReference::createWithRootLoadOrStore(cg, node);
     tempMR->validateImmediateOffsetAlignment(node, size, cg);
 
-    generateTrg1MemInstruction(cg, op, node, targetReg, tempMR);
+    if (needSync && tempMR->getUnresolvedSnippet() == NULL) {
+        TR_ASSERT_FATAL_WITH_NODE(node, (size == 1) || (size == 2) || (size == 4) || (size == 8),
+            "operand size for ldar must be 1, 2, 4, or 8 bytes");
```

Comment:
There is a call to `commonLoadEvaluator()` for loading a 16-byte vector.  Is it safe to assume the load does not require synchronization?
https://github.com/eclipse-omr/omr/blob/aa214b485f6446741a1caf5855ed0d38bc6c5b89/compiler/aarch64/codegen/OMRTreeEvaluator.cpp#L5616-L5619